stock_available_mrp/models/product_product.py

```diff
@@ -0,0 +1,150 @@
+        # compute qty for product with bom
+        product_with_bom = self.filtered('bom_id')
+
+        if not product_with_bom:
+            return res, stock_dict
+        icp = self.env['ir.config_parameter']
+        stock_available_mrp_based_on = icp.sudo().get_param(
+            'stock_available_mrp_based_on', 'qty_available'
+        )
+
+        # explode all boms at once
+        exploded_boms = product_with_bom._explode_boms()
+
+        # extract the list of product used as bom component
+        component_products = self.env['product.product'].browse()
+        for exploded_components in exploded_boms.values():
+            for bom_component in exploded_components:
+                component_products |= first(bom_component).product_id
+
+        # Compute stock for product components.
+        # {'productid': {field_name: qty}}
+        if res and stock_available_mrp_based_on in list(res.values())[0]:
+            # If the qty is computed by the same method use it to avoid
+            # stressing the cache
+            component_qties, _ = \
+                component_products._compute_available_quantities_dict()
+        else:
+            # The qty is a field computed by an other method than the
+            # current one. Take the value on the record.
+            component_qties = {
+                p.id: {
+                    stock_available_mrp_based_on: p[
+                        stock_available_mrp_based_on]} for p in
+                component_products}
+
+        for product in product_with_bom:
+            # Need by product (same product can be in many BOM lines/levels)
+            exploded_components = exploded_boms[product.id]
+            component_needs = product._get_components_needs(
+                exploded_components)
+            if not component_needs:
+                # The BoM has no line we can use
+                potential_qty = 0.0
+
+            else:
+                # Find the lowest quantity we can make with the stock at hand
+                components_potential_qty = min(
+                    [component_qties[component.id][
+                        stock_available_mrp_based_on] / need
+                     for component, need in component_needs.items()]
+                )
+
+                bom_id = product.bom_id
+                potential_qty = (bom_id.product_qty * components_potential_qty)
+                potential_qty = potential_qty > 0.0 and potential_qty or 0.0
+
+                # We want to respect the rounding factor of the potential_qty
+                # Rounding down as we want to be pesimistic.
+                potential_qty = bom_id.product_uom_id._compute_quantity(
+                    potential_qty,
+                    product.bom_id.product_tmpl_id.uom_id,
+                    rounding_method='DOWN'
+                )
+
+            res[product.id]['potential_qty'] = potential_qty
+            res[product.id]['immediately_usable_qty'] += potential_qty
```

Comment:
I not sure of this...
![Selección_302](https://user-images.githubusercontent.com/7689807/60875499-4ae78300-a23a-11e9-9217-1e384936822d.png)
I have a bom for a glass cookcing table
![Selección_301](https://user-images.githubusercontent.com/7689807/60875585-736f7d00-a23a-11e9-8aec-8f334c73967f.png)
Also, I create a work order like this
![Selección_303](https://user-images.githubusercontent.com/7689807/60875618-897d3d80-a23a-11e9-8bd1-811d0861bc89.png)
As yo can see... 
inmediately_usable_qty is 1 because is virtual_available and you increase all potential_qty so inmediately_usable_qty is 6.. Is correct?

Reply:
hello @sergio-teruel 
Yes it is the behavior of the module, I did not change this.
The potential quantity is added to the immediately_usable_qty and it is intended.